### PR TITLE
DROOLS-3122: [DMN Designer] Properties Panel have to expanded before accessing expression editor

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-3122

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/canvas/controls/selection/DomainObjectAwareLienzoMultipleSelectionControl.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import org.kie.workbench.common.stunner.core.client.canvas.event.selection.DomainObjectSelectionEvent;
+import org.kie.workbench.common.stunner.core.domainobject.DomainObject;
+import org.kie.workbench.common.stunner.core.graph.Element;
+
+import static org.kie.soup.commons.validation.PortablePreconditions.checkNotNull;
+
+@Dependent
+@MultipleSelection
+@DMNEditor
+public class DomainObjectAwareLienzoMultipleSelectionControl<H extends AbstractCanvasHandler> extends LienzoMultipleSelectionControl<H> {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/canvas/controls/selection/DomainObjectAwareLienzoMultipleSelectionControl.java]: Would you mind to add some javadoc either for this or parent class?

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/select/MapSelectionControl.java [hunk 2]
```diff
@@ -149,6 +153,39 @@ public boolean isSelected(final Element element) {
...
+            final Element<? extends Definition<?>> element = getElement(selectedItemUUID);
+            return Optional.ofNullable(element);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private Element<? extends Definition<?>> getElement(final String uuid) {
+        return (null != uuid && null != getCanvasHandler()) ? getCanvasHandler().getGraphIndex().get(uuid) : null;
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/select/MapSelectionControl.java]: I think we should move this into `org.kie.workbench.common.stunner.core.client.canvas.util.CanvasLayoutUtils#getElement` and update it accroding to this defensive version.

Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/controls/select/MapSelectionControl.java]: Moved - thanks for the pointer :-)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/canvas/controls/selection/DomainObjectAwareLienzoMultipleSelectionControl.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/canvas/controls/selection/DomainObjectAwareLienzoMultipleSelectionControl.java]: Done.

**Comment by manstis** [comment]
@jomarko Updates made in new commit.

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-forms/kie-wb-common-stunner-forms-client/src/main/java/org/kie/workbench/common/stunner/forms/client/widgets/FormsCanvasSessionHandler.java [hunk 3]
```diff
@@ -262,33 +248,49 @@ private RenderMode getRenderMode(final ClientSession session) {
...
-            renderer.render(getDiagram().getGraph().getUUID(), domainObject);
+            getGraphUUID().ifPresent(graphUUID -> renderer.render(graphUUID,
+                                                                  domainObject,
+                                                                  () -> {/*Nothing*/}));
+        }
+    }
+
+    private void render(final DomainObject domainObject,
+                        final Command callback) {
+        if (null != renderer) {
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-forms/kie-wb-common-stunner-forms-client/src/main/java/org/kie/workbench/common/stunner/forms/client/widgets/FormsCanvasSessionHandler.java]: Updated to be consistent (in this class at least!)

Stunner tends to favour `null != xyz`, I tended to favour `xyz != null` and now we're promoting use of `!Objects.isNull(xyz)`. I'm attempting to use `Objects` static methods where possible ongoing.

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-forms/kie-wb-common-stunner-forms-client/src/main/java/org/kie/workbench/common/stunner/forms/client/widgets/FormsCanvasSessionHandler.java [hunk 4]
```diff
@@ -262,33 +248,49 @@ private RenderMode getRenderMode(final ClientSession session) {
...
+        final Diagram diagram = getDiagram();
+        if (Objects.isNull(diagram)) {
+            return Optional.empty();
+        }
+        final Graph graph = diagram.getGraph();
+        if (Objects.isNull(graph)) {
+            return Optional.empty();
+        }
+        return Optional.of(graph.getUUID());
+    }
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-forms/kie-wb-common-stunner-forms-client/src/main/java/org/kie/workbench/common/stunner/forms/client/widgets/FormsCanvasSessionHandler.java]: Done.

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-forms/kie-wb-common-stunner-forms-client/src/main/java/org/kie/workbench/common/stunner/forms/client/widgets/FormsCanvasSessionHandler.java [hunk 5]
```diff
@@ -129,29 +127,15 @@ public void show() {
...
-                    final String cRoot = diagram.getMetadata().getCanvasRootUUID();
-                    // Check if there exist any canvas root element.
-                    if (!isEmpty(cRoot)) {
-                        selectedItemUUID = cRoot;
+            if (!Objects.isNull(selectionControl)) {
+                selectionControl.getSelectedItemDefinition().ifPresent(selectedItemDefinition -> {
+                    if (selectedItemDefinition instanceof Element) {
+                        render((Element) selectedItemDefinition, callback);
+                    } else if (selectedItemDefinition instanceof DomainObject) {
+                        render((DomainObject) selectedItemDefinition, callback);
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-forms/kie-wb-common-stunner-forms-client/src/main/java/org/kie/workbench/common/stunner/forms/client/widgets/FormsCanvasSessionHandler.java]: Updated the comment here to reference `SelectionControl.getSelectedItemDefinition()` that defines _what_ is considered to be "selected" and shown in the Properties Panel.

**Pull request merged**